From baa647312016cd68db86403aca0b64c60a816e6e Mon Sep 17 00:00:00 2001
From: Haim Boot <hayim@marvell.com>
Date: Tue, 8 Nov 2016 16:13:55 +0200
Subject: [PATCH 0563/1345] dts: watchdog: ap806: added node for watchdog
 configuration

commit  46f5768bc58d1070e4220c167b94826e4c3e65ac from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

this dt entry enables the watchdog which will trigger system
reset upon timeout (such as system hang).

NOTE: WD is not enabled by default.
to enable run 'watchdog -T 10 -t 5 /dev/watchdog'

Change-Id: I9470bac41c19578d92e3549446252bd1294ad798
Signed-off-by: Haim Boot <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33622
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 134fe94..74ee110 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -143,6 +143,14 @@
 				interrupts = <GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
 			};
 
+			watchdog@2a440000 {
+				compatible = "arm,sbsa-gwdt";
+				reg = <0x610000 0x1000>,
+				      <0x600000 0x1000>;
+				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
+				timeout-sec = <30>;
+			};
+
 			revision-info@610fcc {
 				compatible = "marvell,ap806-rev-info";
 				reg = <0x610FCC 0x4>;
-- 
1.7.9.5

